pysat/_instrument.py

```diff
@@ -831,25 +1024,53 @@ def load(self, yr=None, doy=None, date=None, fname=None, fid=None,
-                if len(padLeft) > 0:
-                    if (padLeft.index[-1] == self.data.index[0]) :
-                        padLeft = padLeft.iloc[:-1, :]
-                    self.data = pds.concat([padLeft, self.data])
-
-            if (not self._next_data.empty) & (not self.data.empty):
-                padRight = self._next_data.loc[self.data.index[-1] : last_pad]
-                if len(padRight) > 0:
-                    if (padRight.index[0] == self.data.index[-1]) :
-                        padRight = padRight.iloc[1:, :]
-                    self.data = pds.concat([self.data, padRight])
+            # if (not self._empty(self._prev_data)) & (not self.empty):
+            #     padLeft = self._prev_data.loc[first_pad : self.index[0]]
+            #     if len(padLeft) > 0:
+            #         if (padLeft.index[-1] == self.index[0]) :
+            #             padLeft = padLeft.iloc[:-1, :]
+            #         # self.data = self.concat_data([padLeft, self.data])
+            #     print (padLeft)
+            if (not self._empty(self._prev_data)) & (not self.empty):
+                stored_data = self.data #.copy()
+                temp_time = copy.deepcopy(self.index[0])
+                # pad data using access mechanisms that works
+                # for both pandas and xarray
+                self.data = self._prev_data.copy()
+                self.data = self[first_pad : temp_time]
+                if not self.empty:
+                    if (self.index[-1] == temp_time) :
+                        self.data = self[:-1]                
+                    self.data = self.concat_data([self.data, stored_data])
+                else:
+                    self.data = stored_data
+            #         
+            # if (not self._empty(self._next_data)) & (not self.empty):
+            #     padRight = self._next_data.loc[self.index[-1] : last_pad]
+            #     if len(padRight) > 0:
+            #         if (padRight.index[0] == self.index[-1]) :
+            #             padRight = padRight.iloc[1:, :]
+            #         self.data = self.concat_data([self.data, padRight])
+            if (not self._empty(self._next_data)) & (not self.empty):
+                stored_data = self.data #.copy()
+                temp_time = copy.deepcopy(self.index[-1])
+                # pad data using access mechanisms that work 
+                # for both pandas and xarray
+                self.data = self._next_data.copy()
+                self.data = self[temp_time : last_pad]
+                if not self.empty:
+                    if (self.index[0] == temp_time) :
+                        self.data = self[1:]
+                    self.data = self.concat_data([stored_data, self.data])
+                else:
+                    self.data = stored_data
+                
                     
-            self.data = self.data.ix[first_pad : last_pad]
+            self.data = self[first_pad : last_pad]
```

Comment:
Would it be clearer here to use:

self.data = self.data[first_pad:last_pad]

?

Reply:
Yes, it would be clearer. However, differences in data access details between pandas and xarray  mean for every data access I need to put in a if pandas, do a, if xarray, do b. Which is no good. I put that logic in the __getitem__ method and it does what it needs to based upon pandas, or xarray. Less readable than when pysat was pandas only, but I think more readable than the alternative when supporting two data formats.

Reply:
Perhaps add a comment to that effect here (since it's the first place it's happening)?  That way when someone else looks at the code they know why.

Reply:
comment added